controllers/suite_test.go

```diff
@@ -99,18 +100,19 @@ var _ = BeforeSuite(func(ctx SpecContext) {
 	})
 	Expect(err).ToNot(HaveOccurred())
 
-	bgCtx, cancel := context.WithCancel(context.Background())
+	var cancel context.CancelFunc
+	suiteCtx, cancel = context.WithCancel(context.Background())
 	go func() {
 		defer GinkgoRecover()
 		GinkgoHelper()
 
-		if err := mgr.Start(bgCtx); err != nil {
+		if err := mgr.Start(suiteCtx); err != nil {
 			Fail(fmt.Sprintf("unable to start manager, test can't be ran: %v", err))
 		}
 	}()
 	DeferCleanup(cancel)
 
-	Eventually(mgr.GetCache().WaitForCacheSync).WithContext(ctx).Within(k8sAPIServerResponseTimeout).ProbeEvery(k8sAPIPotentialChangesEvery).Should(BeTrue())
+	Eventually(mgr.GetCache().WaitForCacheSync).WithContext(suiteCtx).Within(k8sAPIServerResponseTimeout).ProbeEvery(k8sAPIPotentialChangesEvery).Should(BeTrue())
```

Comment:
Interesting! I'm surprised that `ctx` gets cancelled prior to the entire test suite completing. Good catch